src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/tests/test_vm_commands.py

```diff
@@ -2385,6 +2386,31 @@ def test_create_vm_from_existing_nic(self, resource_group):
         self.assertTrue(re.match(r'[0-9]+\.[0-9]+\.[0-9]+\.[0-9]+', result['publicIps']))
         self.assertTrue(re.match(r'[0-9]+\.[0-9]+\.[0-9]+\.[0-9]+', result['privateIps']))
 
+
+class VMSecretTest(ScenarioTest):
+    @ResourceGroupPreparer()
+    def test_vm_secret_e2e_test(self, resource_group):
+        # a few pre-made key vault artifacts, with the assumption that this test must under team test subscription of 'AzureSDKADGraph2'
+        keyvault = '/subscriptions/0b1f6471-1bf0-4dda-aec3-cb9272f09590/resourceGroups/sdk-test/providers/Microsoft.KeyVault/vaults/vm-secret-kv'
+        certificate = 'vm-secrt-cert'
+
+        # under playback, keyvault id need to adjust to use mock subscription
+        test_keyvault_id = re.sub('[0-9A-Fa-f]{8}[-][0-9A-Fa-f]{4}[-][0-9A-Fa-f]{4}[-][0-9A-Fa-f]{4}[-][0-9A-Fa-f]{12}', self.get_subscription_id(), keyvault)
+
+        vm = 'vm1'
+        self.cmd('vm create -g {} -n {} --image rhel --use-unmanaged-disk'.format(resource_group, vm))
```

Comment:
If you create the KeyVault and then the VM, I wouldn't think you'd run into DNS issues to issue further KeyVault commands once the VM finished.